Sources/_StringProcessing/Compiler.swift

```diff
@@ -54,3 +54,35 @@ enum RegexCompilationError: Error, CustomStringConvertible {
     }
   }
 }
+
+// Testing support
+@available(SwiftStdlib 5.7, *)
+func _compileRegex(
+  _ regex: String,
+  _ syntax: SyntaxOptions = .traditional,
+  _ semanticLevel: RegexSemanticLevel? = nil
+) throws -> Executor {
+  let ast = try parse(regex, .semantic, syntax)
```

Comment:
Sorry for the merge conflict here, this will need updating to drop the `.semantic` now that https://github.com/apple/swift-experimental-string-processing/pull/519 has been merged